packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -180,8 +194,85 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         electionData.addressVoted[msg.sender] = true;
     }
 
-    function getNomineeVotes(address nominee) external view override returns (uint) {
-        return _electionStore().electionData.nomineeVotes[nominee];
+    function isElectionEvaluated() external view virtual override returns (bool) {
+        return _electionStore().electionData.isElectionEvaluated;
+    }
+
+    function evaluateElectionBatch() external virtual override {
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (_electionStore().electionData.isElectionEvaluated) {
+            revert ElectionAlreadyEvaluated();
+        }
+
+        if (_electionStore().maxProcessingBatchSize == 0) {
+            revert BatchSizeNotSet();
+        }
+
+        _evaluateElectionBatchBySimpleCounting();
+    }
+
+    function _evaluateElectionBatchBySimpleCounting() internal virtual {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
+
+        uint maxBatchSize = store.maxProcessingBatchSize;
+        uint offset = 0;
+        uint previousBatchIdx = electionData.processedBatchIdx;
+        uint lessVotedCandidateVotes;
+        uint lessVotedCandidateIdx;
+
+        while (offset < maxBatchSize && !electionData.isElectionEvaluated) {
+            uint currentNomineeVotes = electionData.nomineeVotes[electionData.nominees[previousBatchIdx + offset]];
```

Comment:
What pushes values into `nomineeVotes` and `winnersVotes`? I see they're only read from in this computation, and I can't find who writes to them.

Reply:
nomineeVotes is filled in elect (while voting), winnersVotes is filled in this loop. There was one line missing though. Fixed.